cmd/darvaza/serve.go

```diff
@@ -10,12 +10,35 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// pr is an interface which is implemented by all proxies
+type pr interface {
+	Run()
+	Cancel() error
+	Reload() error
+}
+
 // Command
 var serveCmd = &cobra.Command{
 	Use:   "serve",
 	Short: "starts serving a proxy",
 	Run: func(cmd *cobra.Command, args []string) {
-		startProxies()
+		// server is just a slice of things implementing
+		// the pr{oxy} interface
+		var server []pr
+		for i := range cfg.Proxies {
+			if z := cfg.Proxies[i].New(); z != nil {
+				server = append(server, z)
+			}
+		}
+		var wg sync.WaitGroup
+		for i := range server {
+			wg.Add(1)
+			go func(i int) {
+				defer wg.Done()
+				server[i].Run()
+			}(i)
+		}
+		wg.Wait()
```

Comment:
this looks wrong. you are blocking before you get the chance to watch for signals

Reply:
fixed in f0a6318b5302e4a333d5a6761231049e80a19b77